rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/ThreadDumpView.java

```diff
@@ -0,0 +1,232 @@
+
+    private RMController controller;
+
+    public ThreadDumpView(RMController controller) {
+        this.controller = controller;
+        RMEventDispatcher eventDispatcher = controller.getEventDispatcher();
+        eventDispatcher.addNodesListener(this);
+        eventDispatcher.addNodeSelectedListener(this);
+    }
+
+    public Canvas build() {
+        VLayout canvas = new VLayout();
+        canvas.setOverflow(Overflow.AUTO);
+
+        createThreadDumpArea();
+        createThreadDumpButtons();
+        createLabels();
+
+        HLayout fetchThreadDumpLayout = new HLayout();
+        fetchThreadDumpLayout.setHeight("20px");
+        fetchThreadDumpLayout.setMembersMargin(10);
+        fetchThreadDumpLayout.addMember(this.nodeThreadDumpButton);
+        fetchThreadDumpLayout.addMember(this.rmThreadDumpButton);
+        fetchThreadDumpLayout.addMember(this.loadingLabel);
+        createLayout(fetchThreadDumpLayout);
+
+        canvas.setMembers(this.noNodeLabel, this.layout);
+        return canvas;
+    }
+
+    public void nodeUnselected() {
+        this.noNodeLabel.setContents("No node selected");
+        this.noNodeLabel.setAlign(Alignment.CENTER);
+        this.noNodeLabel.show();
+        this.layout.hide();
+        this.nodeUrl = null;
+        this.nodeSourceName = null;
+        this.nodeHostName = null;
+    }
+
+    public void nodeSelected(NodeSource.Host.Node node) {
+        this.nodeLabel.setIcon(node.getIcon());
+
+        this.noNodeLabel.hide();
+
+        this.nodeUrl = node.getNodeUrl();
+        this.nodeSourceName = node.getSourceName();
+        this.nodeHostName = node.getHostName();
+        this.nodeLabel.setContents("<h3>" + node.getNodeUrl() + "</h3>");
+        resetThreadDumpArea();
+
+        this.layout.show();
+    }
+
+    public void nodeSourceSelected(NodeSource ns) {
+        this.nodeUrl = null;
+        this.layout.hide();
+        this.noNodeLabel.show();
+    }
+
+    public void hostSelected(NodeSource.Host h) {
+        this.nodeUrl = null;
+        this.layout.hide();
+        this.noNodeLabel.show();
+    }
+
+    @Override
+    public void nodesUpdated(Map<String, NodeSource> nodes) {
+        NodeLabel.update(nodes, this.nodeLabel, this.nodeSourceName, this.nodeHostName, this.nodeUrl);
+    }
+
+    private void createLayout(HLayout executeAndLoading) {
+        this.layout = new VLayout(7);
+        this.layout.setWidth100();
+        this.layout.setHeight100();
+
+        this.layout.addMember(this.nodeLabel);
+        this.layout.addMember(executeAndLoading);
+        this.layout.addMember(this.threadDumpArea);
+
+        this.layout.hide();
+    }
+
+    private void createThreadDumpButtons() {
+        this.nodeThreadDumpButton = new IButton("Fetch Node's Thread Dump");
+        this.nodeThreadDumpButton.setWidth("220px");
+        this.nodeThreadDumpButton.addClickHandler(event -> {
+            if (this.nodeUrl != null) {
+                String nodeUrl = this.nodeUrl;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "nodeUrl" which hides the field declared at line 62. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)